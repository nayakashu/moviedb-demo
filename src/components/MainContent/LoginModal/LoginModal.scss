@import '../../../Common.scss';

.login-form {
  background: $header-bg-color;
  width: 50%;
  margin: auto;
  padding: 50px 80px 65px 80px;
  position: relative;

  form {
    margin-top: 20px;

    input[type='text'],
    input[type='password'] {
      width: 100%;
      background: $body-bg-color;
      border: none;
      padding: 20px;
      margin-bottom: 25px;
      outline: none;
      color: $content-text-color;
    }

    .btn-login {
      outline: none;
      border: none;
      background: $button-blue;
      color: $text-color-light;
      width: 100%;
      padding: 20px;
    }
  }

  .logo {
    display: block;
    text-align: center;

    color: $text-color-light;
    i {
      font-size: 40px;
    }
    .logo-text {
      position: relative;
      top: -14px;
      margin-left: 10px;
    }
  }

  .btn-close {
    color: $content-text-color;
    position: absolute;
    right: 30px;
    top: 25px;
    cursor: pointer;
  }
}
