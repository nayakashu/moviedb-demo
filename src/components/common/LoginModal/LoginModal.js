import React, { Component } from 'react';
import { connect } from 'react-redux';
import { closeLoginForm } from '../../../actions/modalActions';
import './LoginModal.scss';
import makeModal from '../../hocs/makeModal/makeModal';

export class LoginModal extends Component {
  render() {
    return (
      <div className="login-form">
        <span className="logo">
          <i className="material-icons">timeline</i>
          <span className="logo-text">MovieHunt</span>
        </span>

        <i
          className="material-icons btn-close"
          onClick={this.props.closeLoginForm}
        >
          close
        </i>

        <form>
          <input type="text" placeholder="Username" />
          <input type="password" placeholder="Password" />
          <button className="btn-login">Login</button>
        </form>
        <div className="forgot-pass-legend">
          <span className="signup-link">
            Don't have an account? <a href="/#/">Sign up</a>
          </span>
          <span className="recover-pass">Recover password</span>
        </div>
      </div>
    );
  }
}

export default makeModal(
  connect(
    null,
    { closeLoginForm }
  )(LoginModal)
);
