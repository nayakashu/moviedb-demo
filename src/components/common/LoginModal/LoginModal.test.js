import React from 'react';
import { mount } from 'enzyme';
import { LoginModal } from './LoginModal';

describe('LoginModal', () => {
  const props = {
    setUserLogin: jest.fn(),
    closeLoginForm: jest.fn()
  };

  const loginModal = mount(<LoginModal {...props} />);

  it('renders correctly', () => {
    expect(loginModal).toMatchSnapshot();
  });

  it('Simulate login functionality', () => {
    loginModal.find('.btn-login').simulate('click');
    expect(props.setUserLogin).toHaveBeenCalled();
    expect(props.closeLoginForm).toHaveBeenCalled();
  });
});
