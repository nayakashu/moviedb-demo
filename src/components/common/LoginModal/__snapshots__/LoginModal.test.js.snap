// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginModal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="logo"
        >
          <i
            className="material-icons"
          >
            timeline
          </i>
          <span
            className="logo-text"
          >
            MovieHunt
          </span>
        </span>,
        <i
          className="material-icons btn-close"
        >
          close
        </i>,
        <form>
          <input
            placeholder="Username"
            type="text"
          />
          <input
            placeholder="Password"
            type="password"
          />
          <button
            className="btn-login"
            onClick={[Function]}
          >
            Login
          </button>
        </form>,
        <div
          className="forgot-pass-legend"
        >
          <span
            className="signup-link"
          >
            Don't have an account? 
            <a
              href="/#/"
            >
              Sign up
            </a>
          </span>
          <span
            className="recover-pass"
          >
            Recover password
          </span>
        </div>,
      ],
      "className": "login-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="material-icons"
            >
              timeline
            </i>,
            <span
              className="logo-text"
            >
              MovieHunt
            </span>,
          ],
          "className": "logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "timeline",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "timeline",
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieHunt",
              "className": "logo-text",
            },
            "ref": null,
            "rendered": "MovieHunt",
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "material-icons btn-close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "close",
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              placeholder="Username"
              type="text"
            />,
            <input
              placeholder="Password"
              type="password"
            />,
            <button
              className="btn-login"
              onClick={[Function]}
            >
              Login
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "placeholder": "Username",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "placeholder": "Password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "className": "btn-login",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="signup-link"
            >
              Don't have an account? 
              <a
                href="/#/"
              >
                Sign up
              </a>
            </span>,
            <span
              className="recover-pass"
            >
              Recover password
            </span>,
          ],
          "className": "forgot-pass-legend",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Don't have an account? ",
                <a
                  href="/#/"
                >
                  Sign up
                </a>,
              ],
              "className": "signup-link",
            },
            "ref": null,
            "rendered": Array [
              "Don't have an account? ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up",
                  "href": "/#/",
                },
                "ref": null,
                "rendered": "Sign up",
                "type": "a",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Recover password",
              "className": "recover-pass",
            },
            "ref": null,
            "rendered": "Recover password",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="logo"
          >
            <i
              className="material-icons"
            >
              timeline
            </i>
            <span
              className="logo-text"
            >
              MovieHunt
            </span>
          </span>,
          <i
            className="material-icons btn-close"
          >
            close
          </i>,
          <form>
            <input
              placeholder="Username"
              type="text"
            />
            <input
              placeholder="Password"
              type="password"
            />
            <button
              className="btn-login"
              onClick={[Function]}
            >
              Login
            </button>
          </form>,
          <div
            className="forgot-pass-legend"
          >
            <span
              className="signup-link"
            >
              Don't have an account? 
              <a
                href="/#/"
              >
                Sign up
              </a>
            </span>
            <span
              className="recover-pass"
            >
              Recover password
            </span>
          </div>,
        ],
        "className": "login-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="material-icons"
              >
                timeline
              </i>,
              <span
                className="logo-text"
              >
                MovieHunt
              </span>,
            ],
            "className": "logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "timeline",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "timeline",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieHunt",
                "className": "logo-text",
              },
              "ref": null,
              "rendered": "MovieHunt",
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "close",
            "className": "material-icons btn-close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "close",
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                placeholder="Username"
                type="text"
              />,
              <input
                placeholder="Password"
                type="password"
              />,
              <button
                className="btn-login"
                onClick={[Function]}
              >
                Login
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Username",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "btn-login",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="signup-link"
              >
                Don't have an account? 
                <a
                  href="/#/"
                >
                  Sign up
                </a>
              </span>,
              <span
                className="recover-pass"
              >
                Recover password
              </span>,
            ],
            "className": "forgot-pass-legend",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Don't have an account? ",
                  <a
                    href="/#/"
                  >
                    Sign up
                  </a>,
                ],
                "className": "signup-link",
              },
              "ref": null,
              "rendered": Array [
                "Don't have an account? ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                    "href": "/#/",
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "a",
                },
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Recover password",
                "className": "recover-pass",
              },
              "ref": null,
              "rendered": "Recover password",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
